docs/administrators/logging.rst

```diff
@@ -86,59 +63,98 @@ file can be found at ``/usr/lib/systemd/system/inmanta-server.service``.
-  ExecStart=/usr/bin/inmanta --log-file /var/log/inmanta/server.log --log-file-level 2 --timed-logs server
+  ExecStart=/usr/bin/inmanta --log-file /var/log/inmanta/server.log --log-file-level INFO --timed-logs server
   Restart=on-failure
 
   [Install]
   WantedBy=multi-user.target
 
-The ``--log-file-level`` takes the log-level as an integer, where ``0=ERROR``, ``1=WARNING``, ``2=INFO`` and ``3=DEBUG``.
-
-Apply the changes by reloading the service file and restarting the Inmanta server:
 
 .. code-block:: sh
 
   sudo systemctl daemon-reload
   sudo systemctl restart inmanta-server
 
 
-Log level manually started agent
---------------------------------
+Fine grained configuration
+----------------------------
+For fine grained configuration, `a standard python dict config file <https://docs.python.org/3/library/logging.config.html#logging-config-dictschema>`_ can be passed in via the config file for each component individually:
+
+.. code-block:: toml
+
+    [logging]
+    server = server_log.yml
+    scheduler = scheduler.log.tmpl
+    compiler = compiler.yml
+
+or via a cli option:
+
+.. code-block:: sh
+
+    inmanta --logging-config server_log.yml server
+
+
+The log config has to be either a ``yaml`` file, containing a `python dict config <https://docs.python.org/3/library/logging.config.html#logging-config-dictschema>`_ or a template of a ``yaml`` file. In this case, the file name has to end with ``tmpl``.
+
+The following log-related options can be set in an Inmanta config file:
+
+* :inmanta.config:option:`logging.compiler`: determines the log config for the compiler
+* :inmanta.config:option:`logging.server`: determines the log config for the server
+* :inmanta.config:option:`logging.scheduler`: determines the log config for the scheduler. This is always a template.
+
+:ref:`As command line options,<reference_commands_inmanta>` the following are available:
+
+* ``-v``: When used in combination with a log file, it will force a CLI logger to be loaded on top of the provided configuration
+* ``--logging-config``: Log configuration file for this command, overrides the config option
+* If a config file is loaded, all other course grained configuration options are ignored!
+
+For templated config files, we use pythons f-string syntax.
+For the scheduler, one variable is available in the template: ``{environment}``. This is used to customize the scheduler log files to the environment the scheduler is working for.
+
+Converting to fine grained configuration
+----------------------------------------
+
+A tool is provided to convert the existing course grained configuration into a config file.
+
+For example, to convert the config for a component, take the command you use to start it, then put `print-default-logging-config` before the `server`, `compiler` or `scheduler`:
+
+.. code-block:: sh
+
+    inmanta -c /etc/inmanta/inmanta.cfg --log-file /var/log/inmanta/server.log --log-file-level 2 --timed-logs print_default_logging_config server
```

Comment:
Not necessarily a change request, but I find our docs rather hard to read if you have to scroll horizontally in a code block. In my opinion it would even be nice to have some automatic way to detect these issues, so that we can handle them
```suggestion
    inmanta -c /etc/inmanta/inmanta.cfg \
                    --log-file /var/log/inmanta/server.log \
                    --log-file-level 2 \
                    --timed-logs \
                    print_default_logging_config server
```